File: word-defender/icons/generate-icons.py
"""
Generate PNG icons for Word Defender Chrome Extension
Requires: pip install Pillow
"""

try:
    from PIL import Image, ImageDraw
    import math
    
    def draw_icon(size):
        # Create image with gradient background
        img = Image.new('RGB', (size, size), color='#667eea')
        draw = ImageDraw.Draw(img, 'RGBA')
        
        # Draw background gradient effect
        for y in range(size):
            color_value = int(102 + (118 - 102) * (y / size))
            draw.line([(0, y), (size, y)], fill=(color_value, 126, 234))
        
        center_x = size // 2
        center_y = size // 2
        radius = int(size * 0.3)
        
        # Draw tentacles
        tentacle_color = (255, 107, 107, 200)
        tentacle_width = max(1, int(size * 0.05))
        
        for i in range(6):
            angle = (math.pi * 2 / 6) * i
            start_x = center_x + int(math.cos(angle) * radius)
            start_y = center_y + int(math.sin(angle) * radius)
            end_x = center_x + int(math.cos(angle) * radius * 1.8)
            end_y = center_y + int(math.sin(angle) * radius * 1.8)
            
            draw.line([(start_x, start_y), (end_x, end_y)], 
                     fill=tentacle_color, width=tentacle_width)
        
        # Draw main bacteria body (outer glow)
        glow_radius = int(radius * 1.3)
        draw.ellipse([center_x - glow_radius, center_y - glow_radius,
                     center_x + glow_radius, center_y + glow_radius],
                    fill=(255, 107, 107, 100))
        
        # Draw main bacteria body
        draw.ellipse([center_x - radius, center_y - radius,
                     center_x + radius, center_y + radius],
                    fill=(255, 107, 107))
        
        # Draw highlight
        highlight_radius = int(radius * 0.5)
        highlight_x = center_x - int(radius * 0.3)
        highlight_y = center_y - int(radius * 0.3)
        draw.ellipse([highlight_x - highlight_radius, highlight_y - highlight_radius,
                     highlight_x + highlight_radius, highlight_y + highlight_radius],
                    fill=(255, 255, 255, 100))
        
        # Draw eyes (for larger icons)
        if size >= 48:
            eye_size = int(radius * 0.25)
            eye_left_x = center_x - int(radius * 0.3)
            eye_right_x = center_x + int(radius * 0.3)
            eye_y = center_y - int(radius * 0.2)
            
            # White of eyes
            draw.ellipse([eye_left_x - eye_size, eye_y - eye_size,
                         eye_left_x + eye_size, eye_y + eye_size],
                        fill='white')
            draw.ellipse([eye_right_x - eye_size, eye_y - eye_size,
                         eye_right_x + eye_size, eye_y + eye_size],
                        fill='white')
            
            # Pupils
            pupil_size = int(eye_size * 0.5)
            draw.ellipse([eye_left_x - pupil_size, eye_y - pupil_size,
                         eye_left_x + pupil_size, eye_y + pupil_size],
                        fill='black')
            draw.ellipse([eye_right_x - pupil_size, eye_y - pupil_size,
                         eye_right_x + pupil_size, eye_y + pupil_size],
                        fill='black')
        
        # Draw crosshair (for larger icons)
        if size >= 48:
            crosshair_size = int(size * 0.15)
            ch_x = int(size * 0.75)
            ch_y = int(size * 0.25)
            line_width = max(1, int(size * 0.03))
            
            # Crosshair lines
            draw.line([(ch_x - crosshair_size, ch_y), (ch_x + crosshair_size, ch_y)],
                     fill=(0, 255, 255), width=line_width)
            draw.line([(ch_x, ch_y - crosshair_size), (ch_x, ch_y + crosshair_size)],
                     fill=(0, 255, 255), width=line_width)
            
            # Crosshair circle
            circle_radius = int(crosshair_size * 0.7)
            draw.ellipse([ch_x - circle_radius, ch_y - circle_radius,
                         ch_x + circle_radius, ch_y + circle_radius],
                        outline=(0, 255, 255), width=line_width)
        
        return img
    
    # Generate all three icon sizes
    sizes = [16, 48, 128]
    
    for size in sizes:
        img = draw_icon(size)
        filename = f'icon{size}.png'
        img.save(filename)
        print(f'✓ Generated {filename}')
    
    print('\n🎉 All icons generated successfully!')
    print('You can now install the Chrome extension.')
    
except ImportError:
    print('❌ Error: Pillow library not found')
    print('\nTo generate icons with Python:')
    print('1. Install Python (if not already installed)')
    print('2. Run: pip install Pillow')
    print('3. Run: python generate-icons.py')
    print('\nAlternatively:')
    print('- Open create-icons.html in a browser and download icons manually')
    print('- Or use any image editor to create 16x16, 48x48, and 128x128 PNG files')
except Exception as e:
    print(f'❌ Error generating icons: {e}')
    print('\nPlease use create-icons.html in a browser as an alternative.')

File: word-defender/icons/generate-icons.js
// This script generates PNG icons for the Chrome extension
// Run this with Node.js after installing canvas: npm install canvas

const fs = require('fs');
const { createCanvas } = require('canvas');

function drawIcon(size) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');
  
  // Background gradient (approximated with solid color for simplicity)
  ctx.fillStyle = '#667eea';
  ctx.fillRect(0, 0, size, size);
  
  // Draw bacteria-like circle
  const centerX = size / 2;
  const centerY = size / 2;
  const radius = size * 0.3;
  
  // Main bacteria body
  ctx.fillStyle = '#ff6b6b';
  ctx.beginPath();
  ctx.arc(centerX, centerY, radius, 0, Math.PI * 2);
  ctx.fill();
  
  // Tentacles
  ctx.strokeStyle = '#ff6b6b';
  ctx.lineWidth = size * 0.05;
  ctx.lineCap = 'round';
  
  for (let i = 0; i < 6; i++) {
    const angle = (Math.PI * 2 / 6) * i;
    const startX = centerX + Math.cos(angle) * radius;
    const startY = centerY + Math.sin(angle) * radius;
    const endX = centerX + Math.cos(angle) * radius * 1.8;
    const endY = centerY + Math.sin(angle) * radius * 1.8;
    
    ctx.beginPath();
    ctx.moveTo(startX, startY);
    ctx.lineTo(endX, endY);
    ctx.stroke();
  }
  
  // Inner highlight
  ctx.fillStyle = 'rgba(255, 255, 255, 0.3)';
  ctx.beginPath();
  ctx.arc(centerX - radius * 0.3, centerY - radius * 0.3, radius * 0.5, 0, Math.PI * 2);
  ctx.fill();
  
  // Eyes (for larger icons)
  if (size >= 48) {
    ctx.fillStyle = '#ffffff';
    const eyeSize = radius * 0.25;
    ctx.beginPath();
    ctx.arc(centerX - radius * 0.3, centerY - radius * 0.2, eyeSize, 0, Math.PI * 2);
    ctx.arc(centerX + radius * 0.3, centerY - radius * 0.2, eyeSize, 0, Math.PI * 2);
    ctx.fill();
    
    ctx.fillStyle = '#000000';
    const pupilSize = eyeSize * 0.5;
    ctx.beginPath();
    ctx.arc(centerX - radius * 0.3, centerY - radius * 0.2, pupilSize, 0, Math.PI * 2);
    ctx.arc(centerX + radius * 0.3, centerY - radius * 0.2, pupilSize, 0, Math.PI * 2);
    ctx.fill();
  }
  
  // Crosshair overlay
  if (size >= 48) {
    ctx.strokeStyle = '#00ffff';
    ctx.lineWidth = size * 0.03;
    const crosshairSize = size * 0.15;
    const chX = size * 0.75;
    const chY = size * 0.25;
    
    ctx.beginPath();
    ctx.moveTo(chX - crosshairSize, chY);
    ctx.lineTo(chX + crosshairSize, chY);
    ctx.moveTo(chX, chY - crosshairSize);
    ctx.lineTo(chX, chY + crosshairSize);
    ctx.stroke();
    
    ctx.beginPath();
    ctx.arc(chX, chY, crosshairSize * 0.7, 0, Math.PI * 2);
    ctx.stroke();
  }
  
  return canvas;
}

// Generate icons
try {
  const sizes = [16, 48, 128];
  
  sizes.forEach(size => {
    const canvas = drawIcon(size);
    const buffer = canvas.toBuffer('image/png');
    fs.writeFileSync(`icon${size}.png`, buffer);
    console.log(`Generated icon${size}.png`);
  });
  
  console.log('All icons generated successfully!');
} catch (error) {
  console.error('Error generating icons:', error.message);
  console.log('\nTo generate icons, you need to:');
  console.log('1. Install Node.js');
  console.log('2. Run: npm install canvas');
  console.log('3. Run: node generate-icons.js');
  console.log('\nAlternatively, open create-icons.html in a browser and download the icons manually.');
}
